app/decorators/manageiq/providers/autosde/storage_manager/storage_manager_decorator.rb

```diff
@@ -0,0 +1,35 @@
+Module ManageIQ::Providers
+  class AutoSDE::StorageManagerDecorator < MiqDecorator
+    def self.fonticon
+      'pficon pficon-server'
+    end
+
+    def fileicon
+      "svg/vendor-#{image_name}.svg"
+    end
+
+    def quadicon
+      icon = {
+        :top_left     => {},
+        :top_right    => {},
+        :bottom_left  => {
+          :fileicon => fileicon,
+          :tooltip  => ui_lookup(:model => type)
+        },
+        :bottom_right => QuadiconHelper.provider_status(authentication_status, enabled?)
+      }
+
+      icon[:top_left] = {
+        :text    => t = cloud_volumes.size,
+        :tooltip => n_("%{number} Cloud Volume", "%{number} Cloud Volumes", t) % {:number => t}
+      }
+      icon[:top_right] = {
+        :text    => t = storage_resources.size,
+        :tooltip => n_("%{number} Storage Pool", "%{number} Storage Pools", t) % {:number => t}
+      }
```

Comment:
Can you please move these into the declaration above? You no longer have the conditional adding, so it should be cleaner and more readable this way.

Reply:
I left the `[:middle]` outside of the declaration, because it does have  conditional.